File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0] * len(prices)
    stack = []

    # 각 각 넣다가 앞에 넣은 것보다 작아지면 pop
    # pop 2, push 3
    stack.append(0) # index 넣기
    for i in range(1, len(prices)): 
        # 넣는게 크면 넣음
        if prices[stack[-1]] <= prices[i]:
            stack.append(i)
            
        # 작아짐 -> 뺌 & 기록
        else:
            while stack and prices[stack[-1]] > prices[i]:
                top = stack.pop()
                answer[top] = i - top
            stack.append(i)
    while stack:
        top = stack.pop()
        answer[top] = len(prices) - top - 1

    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        // 스택에 넣어서 
        // time 계산
        // 작은게 들어오면 빼서 time을 저장하기
        int time = 0; int index = 0;
        Stack<Integer> st = new Stack<>();
        
        for(int i = 0 ; i < prices.length ; i ++) {
            int curr = prices[i];
            
            while(!st.isEmpty() && curr < prices[st.peek()]) {
                
                int idx = st.pop();
                answer[idx] = i - idx;
            }
            // 넣기, 인덱스 추가
            
            st.push(i);
        }
       while(!st.isEmpty()) {
           int idx = st.pop();
           answer[idx] = prices.length - 1 - idx;
       }
        
        
        return answer;
    }
}